ui/app/routes/vault/cluster/oidc-callback.js

```diff
@@ -6,7 +6,18 @@ export default Route.extend({
     // left blank so we render the template immediately
   },
   afterModel() {
-    let { auth_path: path, code, state } = this.paramsFor(this.routeName);
+    let queryString = window.location.search;
+    // Check if url is encoded
+    if (this.containsEncodedComponents(queryString)) {
+      queryString = decodeURIComponent(queryString);
+    }
```

Comment:
Is this check necessary? I believe if you use `decodeURIComponent` on a string that is not encoded it will return the same string. For example:
```js
decodeURIComponent('?foo=bar') === decodeURIComponent('%3Ffoo%3Dbar'); // true
```
Unless I'm not accounting for something this could be simplified to:
```js
const queryString = decodeURIComponent(window.location.search);
```

Reply:
Oh got it. Thanks, updated!! 👍